consensus/consensus-types/src/proof_of_store.rs

```diff
@@ -39,14 +42,73 @@ pub struct SignedDigestInfo {
     pub num_bytes: u64,
 }
 
+impl SignedDigestInfo {
+    pub fn new(digest: HashValue, expiration: LogicalTime, num_txns: u64, num_bytes: u64) -> Self {
+        Self {
+            digest,
+            expiration,
+            num_txns,
+            num_bytes,
+        }
+    }
+}
+
+// TODO: implement properly (and proper place) w.o. public fields.
+#[derive(Clone, Debug, Deserialize, Serialize)]
+pub struct SignedDigest {
+    epoch: u64,
+    pub peer_id: PeerId,
+    pub info: SignedDigestInfo,
+    pub signature: bls12381::Signature,
+}
+
+impl SignedDigest {
+    pub fn new(
+        epoch: u64,
+        digest: HashValue,
+        expiration: LogicalTime,
+        num_txns: u64,
+        num_bytes: u64,
+        validator_signer: Arc<ValidatorSigner>,
+    ) -> Result<Self, CryptoMaterialError> {
+        let info = SignedDigestInfo::new(digest, expiration, num_txns, num_bytes);
+        let signature = validator_signer.sign(&info)?;
+
+        Ok(Self {
+            epoch,
+            peer_id: validator_signer.author(),
+            info,
+            signature,
+        })
+    }
+
+    pub fn epoch(&self) -> u64 {
+        self.epoch
+    }
+
+    pub fn verify(
+        &self,
+        peer_id: PeerId,
+        validator: &ValidatorVerifier,
+        quorum_store_enabled: bool,
+    ) -> anyhow::Result<()> {
+        if !quorum_store_enabled {
+            return Err(anyhow::anyhow!(
+                "Quorum store is not enabled locally. Sender: {}, epoch: {}",
```

Comment:
I am not sure we need this as this verify will always be called with quorum_store_enabled = true. See the comment below.